hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/streams/AnalyticsStreamFactory.java

```diff
@@ -95,6 +96,13 @@ public StreamFactoryRequirements factoryRequirements() {
             StreamFactoryRequirements.Requirements.ExpectUnauditedGetRequests);
   }
 
+  @Override
+  protected void serviceStop() throws Exception {
+    super.serviceStop();
+    this.s3SeekableInputStreamFactory.close();
```

Comment:
do these before calling the superclass. I think that's actually a no-op, but good to unwind in order